### PR TITLE
add conformance tests

### PR DESCRIPTION
**Pull request by tthsqe12**
Just looking for comments. @fingolfin @wbhart
~~I like the approach of the user providing explicit elements, which can hit coner cases, and also gives you the elem_type.~~
It uses functions now.
Will work on matrix, and poly tests soon.

Currently these rings do not pass, but the other ones work well.
```
zz
RDF
RealField
ResidueRing(ZZ, 1)
```

### CODE REVIEW

**Comment by wbhart** [comment]
There was previously a very long discussion on a conformance test ticket about zz. @fieker and I were of the opinion that zz only has partial support (especially we should not support put time into making polynomials and matrices over zz pass) and it is not a priority for us as an Oscar ring.

(I will mention for the sake of completeness that others on that ticket disagreed with us.)

File: test/Rings-conformance-tests.jl [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      end
+   end
+end
+
+
+#InexactError on R(BigInt(3) ^ 100)
+#failure on iszero(b) || divexact(b * a, b) == a
+#test_Ring_interface(zz, Int[11, 2^30, 2^40])
+
+test_Ring_interface(ZZ, BigInt[11])
```
Comment (fingolfin) [inline test/Rings-conformance-tests.jl:197]: I would recommend to move the invocations of `test_Ring_interface` to another file, ideally to the files containing the other tests for the ring(s) in question. This way other packages can load this file to be able to use `test_Ring_interface`.

Comment (tthsqe12) [reply test/Rings-conformance-tests.jl]: will do ...

**Comment by tthsqe12** [comment]
If zz is to have support, it should be considered as a fast implementation of ZZ/2^64 or 2^32

**Comment by wbhart** [comment]
I'm not so sure. zz = Int, not UInt. It isn't how we handle residues elsewhere.

**Review by thofma — APPROVED** [review]

**Review by wbhart — APPROVED** [review]

**Pull request merged**